packages/kbn-observable/src/observable.test.ts

```diff
@@ -0,0 +1,986 @@
+    consoleError.mockImplementation(noop);
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+      return cleanup;
+    });
+
+    source.subscribe({
+      error() {
+        throw new Error('thrown from error');
+      },
+    });
+
+    observer.error(new Error('foo'));
+
+    expect(cleanup).toHaveBeenCalledTimes(1);
+    expect(consoleError.mock.calls).toMatchSnapshot('logs');
+
+    consoleError.mockRestore();
+  });
+});
+
+describe('SubscriptionObserver#complete', () => {
+  test('it does not forward arguments', () => {
+    let observer!: SubscriptionObserver<string>;
+    let observerArgs: any;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe({
+      complete(...args: any[]) {
+        observerArgs = args;
+      },
+    });
+
+    (observer as any).complete('foo', 'bar', 'baz');
+
+    expect(observerArgs).toEqual([]);
+  });
+
+  test('suppresses the value returned from the observer', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe({
+      complete() {
+        return 'bar';
+      },
+    });
+
+    expect(observer.complete()).toBe(undefined);
+  });
+
+  test('returns undefined when closed', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe({
+      complete(...args: any[]) {
+        return 'bar';
+      },
+    });
+
+    observer.complete();
+    expect(observer.complete()).toBe(undefined);
+  });
+
+  test('catches errors thrown from the observer', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const consoleError = jest.spyOn(console, 'error');
+    consoleError.mockImplementation(noop);
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    source.subscribe({
+      complete() {
+        throw new Error('thrown from complete');
+      },
+    });
+
+    expect(observer.complete()).toBe(undefined);
+    expect(consoleError.mock.calls).toMatchSnapshot('logs');
+
+    consoleError.mockRestore();
+  });
+
+  test('can call "complete" on observer when no "complete" is defined when subscribed', () => {
+    let observer!: SubscriptionObserver<string>;
+    const cleanup = jest.fn();
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+      return cleanup;
+    });
+
+    source.subscribe();
+
+    expect(observer.complete()).toBe(undefined);
+    expect(cleanup).toHaveBeenCalledTimes(1);
+  });
+
+  test('does not call "complete" multiple times', () => {
+    let observer!: SubscriptionObserver<string>;
+
+    const source = new Observable<string>(_observer => {
+      observer = _observer;
+    });
+
+    const values: any[] = [];
```

Comment:
__optional nit:__ maybe just `const complete = jest.fn();` and `toHaveBeenCalledTimes(1)`?